lib/rails/perftest/active_support/testing/performance/ruby.rb

```diff
@@ -49,9 +49,18 @@ def record
 
           klasses.each do |klass|
             fname = output_filename(klass)
-            FileUtils.mkdir_p(File.dirname(fname))
-            File.open(fname, 'wb') do |file|
-              klass.new(@data).print(file, full_profile_options.slice(:min_percent))
+            printer = klass.new(@data)
+            options = full_profile_options.slice(:min_percent)
+            path = File.dirname(fname)
+            FileUtils.mkdir_p(path)
+            if printer.is_a?(RubyProf::CallTreePrinter) && RubyProf::VERSION >= "0.16"
+              printer.print(options.merge(:path => path))
```

Comment:
`ruby-prof` creates a new file associated with the process id when printing the _CallTree_ (https://github.com/ruby-prof/ruby-prof/blob/e7bd620074df2d3017644566966a05b8a183686d/lib/ruby-prof/printers/call_tree_printer.rb#L50).

The next two lines are considering the hardcoded logic for naming this file (https://github.com/ruby-prof/ruby-prof/blob/e7bd620074df2d3017644566966a05b8a183686d/lib/ruby-prof/printers/call_tree_printer.rb#L105) to rename it after creation, so the files generated by `rails-perftest` under `ruby-prof` < 0.16 or `>= 0.16` are the same.